src/Settings.cpp

```diff
@@ -19,6 +20,62 @@ static bool ends_with(const std::string& str, const std::string& with)
     return str.rfind(with) == str.size() - with.size();
 }
 
+void Settings::setUserConfigurationFile(const QString userConfigurationFileArg)
+{
+    userConfigurationFile = userConfigurationFileArg;
+}
+
+void Settings::setSettingsObject()
+{
+    /*
+    Variable that stores whether or not the configuration file requested by the user is a normal configuration file
+    If the file does not exist and is newly created, the if statement below is not executed, so the default value is set to true
+    */
+    bool isNormalUserConfigurationFile = true;
+
+    // Code that verifies that the configuration file requested by the user is a normal configuration file
+    if(userConfigurationFile != nullptr)
+    {
+        QFile *file = new QFile;
+        file->setFileName(userConfigurationFile);
+
+        if(file->open(QIODevice::ReadOnly))
+        {
+            if(file->exists())
+            {
+                if(QString::compare(QString::fromStdString("[%General]\n"), file->readLine(), Qt::CaseInsensitive) != 0)
```

Comment:
If you're concerned about spaghetti code, you could take advantage of short-circuit evaluation of `&&` and unify all these conditions in one `if` (separated in different lines for readability).